src/core/qgspointlocator.cpp

```diff
@@ -685,7 +695,8 @@ bool QgsPointLocator::rebuildIndex( int maxFeaturesToIndex )
     return true; // nothing to index
 
   QgsFeatureRequest request;
-  request.setSubsetOfAttributes( QgsAttributeList() );
+  // request.setSubsetOfAttributes( QgsAttributeList() );
```

Comment:
I'd leave this, and then in the branch below when a renderer is set you should set the subset of attributes to that required by the renderer.